bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -0,0 +1,283 @@
+        }
+    }
+
+    @Modified
+    @SuppressWarnings("null")
+    protected void modified(ComponentContext componentContext) {
+        try {
+            logger.debug("Shelly Binding Configuration refreshed");
+            bindingConfig.updateFromProperties(componentContext.getProperties());
+        } catch (RuntimeException e) {
+            logger.warn("Exception on ShellyDiscoveryParticipant.modified(): {} ({})", e.getClass(), e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Nullable
+    @Override
+    public DiscoveryResult createResult(@Nullable ServiceInfo service) {
+        Validate.notNull(bindingConfig);
+        if ((service == null) || !service.getName().startsWith("shelly")) {
+            return null;
+        }
+
+        String address = "";
+        String name = "";
+        String mode = "";
+        String model = "unknown";
+        String thingType = "";
+        ThingUID thingUID = null;
+        ShellyDeviceProfile profile = null;
+        Map<String, Object> properties = new HashMap<String, Object>();
+
+        try {
+            name = service.getName().toLowerCase();
+            address = StringUtils.substringBetween(service.toString(), "/", ":80");
+            if (address == null) {
+                logger.debug("Discovered Shelly device {} doesn't have an IP address, service-info={}", name, service);
+                return null;
+            }
+            logger.info("Shelly device discovered: IP-Adress={}, name={}", address, name);
+
+            ShellyThingConfiguration config = new ShellyThingConfiguration();
+            if (handlerFactory != null) {
+                bindingConfig = handlerFactory.getBindingConfig();
+                Validate.notNull(bindingConfig);
+            }
+            config.deviceIp = address;
+            config.userId = bindingConfig.defaultUserId;
+            config.password = bindingConfig.defaultPassword;
+
+            // Get device settings
+            ShellyHttpApi api = new ShellyHttpApi(config);
+            thingType = StringUtils.substringBeforeLast(name, "-");
+
+            try {
+                profile = api.getDeviceProfile(thingType);
+                Validate.notNull(profile);
+                logger.debug("Shelly settings : {}", profile.settingsJson);
+                Validate.notNull(profile, "Unable to get device profile: ");
+                model = profile.settings.device.type;
+                mode = profile.mode;
+
+                properties = ShellyBaseHandler.fillDeviceProperties(profile);
+                logger.trace("name={}, thingType={}, deviceType={}, mode={}", name, thingType, profile.deviceType,
+                        mode.isEmpty() ? "<standard>" : mode);
+            } catch (IOException e) {
+                if (e.getMessage().contains(APIERR_HTTP_401_UNAUTHORIZED)) {
+                    logger.warn("Device {} ({}) reported 'Access defined' (userid/password mismatch).", name, address);
```

Comment:
As discussed, please remove these warning/info messages. The discovery should merely state that there is a device - at this stage, it is not expected that one can authenticate against it.

Reply:
done